### ISSUE TITLE
Pagination issues and considerations

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon-website#3238]
This is an omnibus issue covering the Pagination component, resulting from an accessibility review. In addition to the top items covered here, there is specific and more detailed material covered in commented PDFs of [Usage](https://ibm.box.com/s/6ggcperys16fda1xibr3ideh3oezz9eu) and [Style](https://ibm.box.com/s/ww2z3hmgfuoqkwpj65cdgxfbl6nykwcc) pages that should be reviewed.

- The following sections are missing from the Usage: Formatting, Anatomy, Placement, Content, Behaviors, Related, References. Given the extreme relevance of at least data table and filtering patterns, this seems like a pretty big omission.
- There is a whole pagination section in Data table. i wonder whether that information should be relocated here, or is at least harvested for what content is not unique to a table pagination interaction.
- Note that the data table's indetermine version of pagination does not seem to perfectly align with that shown in the Live demo
- The labelling is not great for a screen reader, and I think there could be ways of making it better. The simplest is:
"Page: [12 of 40]v" instead of "[12]v of 40 pages" I've used the "[ ]v" to represent where the input and down chevron is/could be. That would need to be tested, but with zero real difference to a sighted user could likely be a much better screen reader or magnified result. Note that this also works when the unknown variant is enabled
![image](https://user-images.githubusercontent.com/14298245/199504829-60e5dc30-55b1-49f9-b459-6c0234776c38.png)
![image](https://user-images.githubusercontent.com/14298245/199504860-b0bc93da-02d3-4af6-9822-d337d3932d70.png)
There are more holistic changes that could be made, but this would be a pretty easy one to do.
- the current design also requires you to do something with the 'orphan' text when it comes to building a label. At the moment, Carbon has used a hidden aria label that is populated based on values on screen, but I think an `aria-describedby` is more elegant and also better supports internationalization
`<label for="cds-pagination-select-id-5-right" class="cds--label cdes--visually hidden">Page number, of 6 pages</label>
`
They could just take the select itself and use aria-describedby to link it with an ID they assign to
`<span class"cds--pagination__text">of 6 pages</span>`
The speech output with Jaws is much smoother, IMO. I've made a small recording of [current](https://ibm.box.com/s/tg6h65u1zixebmp5ilc9fz3rh5a6w6gs) and [improved](https://ibm.box.com/s/zdfg0x5hlb0tu9uc2cybgv4tm46x2fwc) screen reader behaviour. If combined with the above suggestion on labelling changes, this would result in real improvement.
- the chunk of text after the first input ("1 to 10 of 103 items" in the image above) should be tested as a live region. The advantages of this: when a user incremented the prior or next pages, or when a user changed the number of records shown OR  potentially when the filtering was altered, this would be announced by the screen reader. Could be VERY useful.
- seem to be unintended duplicates at the bottom of the Usage page
- the page totals used in an illustration on the style page do not make sense
- Speaking of putting "Page" in front of the right-side input, almost all the variations in Figma already show that, so it would result in a more consistent layout
<img width="366" alt="image" src="https://user-images.githubusercontent.com/14298245/199510541-e5733f21-296c-45e7-b176-f71b4132a09c.png">

**Comment by laurenmrice** [comment]
Website docs:
- [ ] Add a section for "Responsiveness" and how the pagination content should change and respond depending on the breakpoint. Only pertinent information should remain in the pagination component at smaller breakpoints.

**Comment by laurenmrice** [comment]
Closing in favor of the following issues where the work for this will be tackled and cross-referenced:
- Website: https://github.com/carbon-design-system/carbon-website/issues/4013
- Website: https://github.com/carbon-design-system/carbon-website/issues/4021
- Code: https://github.com/carbon-design-system/carbon/issues/16538
